Mathlib/Algebra/Ring/Divisibility/Lemmas.lean

```diff
@@ -91,3 +92,32 @@ lemma add_pow_dvd_pow_of_pow_eq_zero_left (hy : y ^ n = 0) :
 end Ring
 
 end Commute
+section CommRing
+
+variable [CommRing R]
+
+lemma dvd_mul_sub_mul_mul_left_of_dvd {p a b c d x y : R}
+    (h1 : p ∣ a * x + b * y) (h2 : p ∣ c * x + d * y) : p ∣ (a * d - b * c) * x := by
+  obtain ⟨k1, hk1⟩ := h1
+  obtain ⟨k2, hk2⟩ := h2
+  refine ⟨d * k1 - b * k2, ?_⟩
+  rw [show (a * d - b * c) * x = a * x * d - c * x * b by ring, eq_sub_of_add_eq hk1,
+    eq_sub_of_add_eq hk2]
+  ring
+
+lemma dvd_mul_sub_mul_mul_right_of_dvd {p a b c d x y : R}
+    (h1 : p ∣ a * x + b * y) (h2 : p ∣ c * x + d * y) : p ∣ (a * d - b * c) * y := by
+  obtain ⟨k1, hk1⟩ := h1
+  obtain ⟨k2, hk2⟩ := h2
+  refine ⟨-c * k1 + a * k2, ?_⟩
+  rw [show (a * d - b * c) * y = d * y * a - b * y * c by ring, eq_sub_of_add_eq' hk1,
+    eq_sub_of_add_eq' hk2]
+  ring
```

Comment:
```suggestion
lemma dvd_mul_sub_mul_mul_right_of_dvd {p a b c d x y : R}
    (h1 : p ∣ a * x + b * y) (h2 : p ∣ c * x + d * y) : p ∣ (a * d - b * c) * y :=
  (mul_comm a _ ▸ mul_comm c _ ▸ dvd_mul_sub_mul_mul_left_of_dvd 
    (add_comm (c * x) _ ▸ h2) (add_comm (a * x) _ ▸ h1))
```
What do you think about deducing it from the previous lemma? This proof is about twice as fast and doesn't involve any fancy tactics like `ring`, but I'm ambivalent about whether it's better.